dns-server/src/dns_server.rs

```diff
@@ -268,6 +268,19 @@ async fn handle_dns_message(
     let mut additional_records = vec![];
     let response_records = records
         .into_iter()
+        .filter(|record| {
```

Comment:
I think this is fine but I'm curious why you decided to filter here and not in the store?

Reply:
mostly expecting that we would reasonably want related records in other cases. an immediate example might be returning AAAA records as additionals for an A query, or vice versa (admittedly tenuous, since there's not much reason A records would be anywhere.)

maybe more reasonably: this makes it a bit easier to detect (and warn?) about clients doing AAAA queries directly to the name for a SRV record.

if we weren't storing all the records for a name together, and filtering in the store could have saved some work, i'd be strongly in favor of filtering there.